### ISSUE TITLE
Low final contrast accuracy value

### ISSUE THREAD

**Issue by CarolRameder** [djimenezsanchez/NaroNet#4]
Hello! 

I am working with Naronet on my dissertation thesis under the supervision of @miselico.
I have tried to run Naronet on the following databases: example_POLE, halved example_POLE and Endometrial POLE (full dataset).
Each time I got a low final contrast accuracy value (under 2%), even though I followed the steps from the Readme file. The only change I made is limiting the model to one GPU. I modified the next line in NaroNet.py->class NaroNet-> init-> 2nd line 
self.device = 'cuda:0'. This created a runtime error as only the computer I am using only has one GPU available.  

As mentioned in the README, the accuracy should be way higher to continue using the model components:
"To check whether the CNN has been trained successfully, check the 'Model_training_xxxx' folder and open the 'Contrast_accuracy_plot.png', where you should expect a final contrast accuracy value over 50%."

I attached the plot generated in my case. How can this issue be solved? 
Thanks in advance!

![Contrast_accuracy_plot](https://github.com/djimenezsanchez/NaroNet/assets/22241657/49fc8e65-3832-45bb-95d1-88c93cffb662)

**Comment by djimenezsanchez** [comment]
Hi, 

Thanks for your message. Which parameters did you use to run PCL? 

The parameters used for the paper's results were tested on the endometrial's full dataset and are the following:

![image](https://github.com/djimenezsanchez/NaroNet/assets/43730952/4d5eadc1-ceb7-4375-b37f-fee79d1169e9)

To increase the performance you can try increasing the batch size. This is the most important parameter, usually limited by the GPU size.

Best
Daniel

**Comment by CarolRameder** [comment]
Hi,

Thanks for your response! I am coming back with some updates 
I have the following results:

Example Pole
Batch size  = 160 -> final accuracy = 0.63%
Batch size  = 800 -> final  accuracy = 0.12%; stopped after only 1750 steps instead of 20k as for the others
Batch size  = 400 -> final accuracy = 0.25%
Batch size  = 20 -> final  accuracy = 5% 

SyntheticCCI1
Batch size  = 160 ->  final accuracy = 0.63%
Batch size  = 20 ->  final accuracy = 5%

Endometrial POLE
Batch size = 80 -> final accuracy = 6%

The parameters used for the Example POLE and Endometrial POLE were those pre-set in the DatasetParameters.py for the Endometrial POLE case (I created a new elif statement for the Example POLE and pasted the same parameters). 
For the SyntheticCCI1, I used the default parameters in the last "else" statement in DatasetParameters.py. Can you suggest how the parameters predefined in "SyntheticV1" to "SyntheticV4" should be used (and perhaps SyntheticV_H1-4 as well)?

Currently, I am using this system: 
Ubuntu 22.04.2: python==3.8, pytorch=1.13.1, cudatoolkit=11.7.0, nvidia-cudnn-cu11==8.2 tensorflow==2.10; 2x Nvidia RTX A6000

Previously I have used the following system and got similar performance: 
WSL2 on Windows10: python=3.7.3, pytorch==1.13.1, pytorch-cuda=11.7, tensorflow=1.15.8 (directml);  Nvidia GeForce RTX 3060

Is there any other possible reason behind this gap in accuracy for the PCL component if the experimental setup is followed step-by-step as in the readme page of the repository?
The only conclusion that I can draw is that increasing the batch size seems to affect the final accuracy and that the PCL component does not seem to learn good embeddings at all, no matter the dataset or system used.

**Comment by miselico** [comment]
I do not get the closing of the issue as "not planned". This is not a request for additional implementation, but rather a question on how to reproduce the results of the paper.
